File: src/GS.Certifications.Application/Commons/Dtos/OrdenesCompra/OrdenCompraDto.cs
using AutoMapper;
using GS.Certifications.Application.Commons.Dtos.Empresas;
using GSF.Application.Common.Mappings;
using GSF.Application.Global.Currencies.Queries;
using GSF.Application.Security.Services;
using GS.Certifications.Domain.Entities.OrdenesCompras;
using System;

namespace GS.Certifications.Application.Commons.Dtos.OrdenesCompra
{
    public class OrdenCompraDto : IMapFrom<OrdenCompra>
    {
        public int Id { get; set; }
        public string NumeroOrden { get; set; }
        public DateTime Fecha { get; set; }
        public EmpresaPortalDto EmpresaPortal { get; set; }
        public OrdenCompraTipoDto OrdenCompraTipo { get; set; }
        //public string CodigoHES { get; set; }
        public string CodigoQAD { get; set; }
        public OrdenCompraEstadoDto OrdenCompraEstado { get; set; }
        public decimal Importe { get; set; } = default;
        public CurrencyDto Moneda { get; set; }
        public string Observaciones { get; set; }
        public SecurityUserCompaniesDto Company { get; set; }

        public class MappingProfile : Profile
        {
            public MappingProfile()
            {
                CreateMap<OrdenCompra, OrdenCompraDto>()
                .ForMember(dest => dest.EmpresaPortal, opt => opt.MapFrom(src => src.EmpresaPortal))
                .ForMember(dest => dest.OrdenCompraTipo, opt => opt.MapFrom(src => src.OrdenCompraTipo))
                .ForMember(dest => dest.OrdenCompraEstado, opt => opt.MapFrom(src => src.OrdenCompraEstado))
                .ForMember(dest => dest.Moneda, opt => opt.MapFrom(src => src.Moneda))
                .ForMember(dest => dest.Company, opt => opt.MapFrom(src => src.Company));

                CreateMap<OrdenCompraTipo, OrdenCompraTipoDto>();
                CreateMap<OrdenCompraEstado, OrdenCompraEstadoDto>();
            }
        }
    }
}

File: src/GS.Certifications.Web/wwwroot/src/Proveedores/OrdenesCompras/OrdenCompraDto.js
export default class OrdenCompraDto {
    constructor(dto) {
        this.id = dto ? dto.id : null;
        this.rowVersion = dto ? dto.rowVersion : null;
        this.numeroOrden = dto ? dto.numeroOrden : null;
        this.fecha = dto ? dto.fecha : null;
        this.empresaPortal = dto ? dto.empresaPortal : null;
        this.ordenCompraTipo = dto ? dto.ordenCompraTipo : null;
        //this.codigoHES = dto ? dto.codigoHES : null;
        this.codigoQAD = dto ? dto.codigoQAD : null;
        this.ordenCompraEstado = dto ? dto.ordenCompraEstado : null;
        this.importe = dto ? dto.importe : null;
        this.moneda = dto ? dto.moneda : null;
        this.observaciones = dto ? dto.observaciones : null;
    }
}